src/dmd/root/string.d

```diff
@@ -212,3 +212,82 @@ unittest
     assert(dstrcmp("안녕하세요!", "안녕하세요!") == 0);
     static assert(dstrcmp("안녕하세요!", "안녕하세요!") == 0);
 }
+
+/**
+ * Infers the length `N` of a string literal and coerces its type to a static
+ * array with length `N + 1`. Returns the string with a null character appended
+ * to the end.
+ *
+ * Params:
+ *  literal = string literal
+ *
+ * Notes:
+ *  - LDC produces quite optimal code for short strings:
+ *    https://d.godbolt.org/z/mffrjy,
+ *    https://gist.github.com/PetarKirov/338e4ab9292b6b2b311a3070572a07fb (backup URL)
+*/
+T[N + 1] toStaticArray(T, size_t N)(scope const(T)[N] literal)
```

Comment:
Since this is never called, why is it here?

Reply:
I see it's called in the unittest. It just seems excessive for what it does.

Reply:
As mentioned elsewhere, I plan to use it in other parts of the codebase, in order to reduce the need for heap allocations and make the code more concise.

Reply:
It's over-engineering.